### PR TITLE
Adapt to Submissions update

### PR DESCRIPTION
**Pull request by siemensikkema**
- setup to setUp where necessary

### CODE REVIEW

File: Sources/App/Models/Entities/AppUser/Extensions/AppUser+JWTKeychain.swift [hunk 1]
```diff
@@ -4,78 +4,74 @@ import Reset
...
 
     static let usernameKey: WritableKeyPath<AppUser, String> = \.email
     static let passwordKey: WritableKeyPath<AppUser, String> = \.password
+}
 
-    struct UserLogin: HasReadablePassword, HasReadableUsername {
-        static let readablePasswordKey = \UserLogin.password
-        static let readableUsernameKey = \UserLogin.email
+// MARK: - Creatable
+extension AppUser {
```
Comment (steffendsommer) [inline Sources/App/Models/Entities/AppUser/Extensions/AppUser+JWTKeychain.swift]: Any reason for why the conformance is not added to the type?

File: Sources/App/Models/Entities/AppUser/Extensions/AppUser+JWTKeychain.swift [hunk 2]
```diff
@@ -4,78 +4,74 @@ import Reset
...
+            password: AppUser.hashPassword(create.password)
+        )
     }
+}
 
-    struct UserRegistration: HasReadablePassword, HasReadableUsername {
-        static let readablePasswordKey = \UserRegistration.password
-        static let readableUsernameKey = \UserRegistration.email
+// MARK: - Loginnable
+extension AppUser {
```
Comment (steffendsommer) [inline Sources/App/Models/Entities/AppUser/Extensions/AppUser+JWTKeychain.swift]: Any reason for why the conformance is not added to the type?

File: Sources/App/Models/Entities/AppUser/Extensions/AppUser+JWTKeychain.swift [hunk 3]
```diff
@@ -4,78 +4,74 @@ import Reset
...
-
-        let email: String?
-        let name: String?
-        let password: String?
-        let oldPassword: String?
-
-        var username: String? {
-            return email
-        }
+// MARK: - PublicRepresentable
```
Comment (steffendsommer) [inline Sources/App/Models/Entities/AppUser/Extensions/AppUser+JWTKeychain.swift]: Any reason for why the conformance is not added to the type?

File: Sources/App/Models/Entities/AppUser/Extensions/AppUser+JWTKeychain.swift [hunk 4]
```diff
@@ -4,78 +4,74 @@ import Reset
...
+    func convertToPublic(on req: Request) throws -> Future<Public> {
+        return req.future(Public(email: email, name: name))
     }
+}
 
-    func convertToPublic(on req: Request) throws -> Future<UserPublic> {
-        return req.future(UserPublic(email: email, name: name))
-    }
+// MARK: - Updatable
+extension AppUser {
```
Comment (steffendsommer) [inline Sources/App/Models/Entities/AppUser/Extensions/AppUser+JWTKeychain.swift]: Any reason for why the conformance is not added to the type?

File: Sources/App/Models/Entities/AppUser/Extensions/AppUser+Submittable.swift [hunk 5]
```diff
@@ -4,78 +4,76 @@ import Submissions
...
                     validators: [.count(8...)]
                 ),
-                makeFieldEntry(
-                    keyPath: \Submission.passwordAgain,
+                Field(
+                    keyPath: \.passwordAgain,
+                    instance: instance,
                     label: "Password again",
-                    validators: [.count(8...)]
+                    validators: [Validator("") {
```
Comment (steffendsommer) [inline Sources/App/Models/Entities/AppUser/Extensions/AppUser+Submittable.swift:33]: 👍

File: Sources/App/Setup/configure.swift [hunk 6]
```diff
@@ -67,12 +66,7 @@ public func configure(
     services.register(OffsetPaginatorConfig.current)
 
     // MARK: Leaf tags
-
-    services.register { container -> LeafTagConfig in
-        var config = LeafTagConfig.default()
-        try leafTags(config: &config, container)
-        return config
-    }
+    // Look at boot.swift
```
Comment (steffendsommer) [inline Sources/App/Setup/configure.swift]: We're registering leaf tags in `Setup/leafTags.swift` (https://github.com/nodes-vapor/template/blob/feature/adapt-to-submissions/Sources/App/Setup/leafTags.swift) so this function needs to be called. Also, is it needed to do it through `boot.swift` still? I believe we did that due to the (now outdated) `MutableLeafTagContainer`.

**Review by steffendsommer — CHANGES_REQUESTED** [review]

File: Sources/App/Setup/configure.swift (hunk 6 above)
Comment (siemensikkema) [reply Sources/App/Setup/configure.swift]: do you have an example of how this should be done?

Comment (siemensikkema) [reply Sources/App/Setup/configure.swift]: never mind 🙂

**Pull request merged**